source/org/zfin/uniprot/secondary/SecondaryTermLoadAction.java

```diff
@@ -0,0 +1,153 @@
+            case UNIPROTKB ->  "UniProtKB-KW:" + accession;
+            case EC -> "EC:" + accession;
+            default ->  dbName.toString() + ":" + accession;
+        };
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof SecondaryTermLoadAction that)) return false;
+        return compareTo(that) == 0;
+    }
+
+    @Override
+    public int compareTo(SecondaryTermLoadAction o) {
+        Comparator<SecondaryTermLoadAction> comparator = Comparator.comparing(
+                        (SecondaryTermLoadAction obj) -> obj.accession, ObjectUtils::compare)
+                .thenComparing(obj -> obj.type, ObjectUtils::compare)
+                .thenComparing(obj -> obj.subType, ObjectUtils::compare)
+                .thenComparing(obj -> obj.geneZdbID, ObjectUtils::compare)
+                .thenComparing(obj -> obj.details, ObjectUtils::compare)
+                ;
+        return comparator.compare(this, o);
+    }
+
+    public String toString() {
+        return "InterproLoadAction: " + " action=" + type +
+                " subtype=" + subType +
+                " accession=" + accession +
+                " goID=" + goID +
+                " goTermZdbID=" + goTermZdbID +
+                " geneZdbID=" + geneZdbID +
+                " relatedEntityID=" + relatedEntityID +
+                " details=" + details +
+                " length=" + length +
+                " handlerClass=" + handlerClass +
+                " relatedEntityFields=" + relatedEntityFieldsToString() +
+                " links=" + links;
+    }
+
+    public String markerGoTermEvidenceRepresentation() {
+        return geneZdbID + "," + goTermZdbID + "," + goID + "," + dbName + ":" + this.accession;
+    }
+
+    public String getMd5() {
+        Md5PasswordEncoder encoder = new Md5PasswordEncoder();
+        return encoder.encodePassword(toString(), null);
+    }
+
+    public void setMd5(String md5) {
+        //purposefully empty for deserialization
+    }
+
+    @JsonIgnore
+    public String relatedEntityFieldsToString() {
+        if (relatedEntityFields == null) {
+            return "";
+        }
+        StringBuilder sb = new StringBuilder("{");
+        for (Map.Entry<String, String> entry : relatedEntityFields.entrySet()) {
+            sb.append(entry.getKey()).append("=").append(entry.getValue()).append(";");
+        }
+        sb.append("}");
+        return sb.toString();
+    }
+
+    public List<UniProtLoadLink> getDynamicLinks() {
+        List<UniProtLoadLink> dynamicLinks = new ArrayList<>();
+        if ( accession != null ) {
+            if (dbName != null && dbName.equals(ForeignDB.AvailableName.INTERPRO)) {
+                dynamicLinks.add(new UniProtLoadLink(accession, "http://www.ebi.ac.uk/interpro/entry/" + accession));
+            }
+        }
+        if ( goID != null ) {
+            dynamicLinks.add(new UniProtLoadLink(goID, "https://zfin.org/" + goID));
```

Comment:
Why hard-coding zfin.org into calling the constructor? Should be hard-coded somewhere else.WIll there ever be a different value?